File: cases/taint/java/local-overwrite-negative/LocalOverwriteNegative.java
package dataflowbench.taint;

final class LocalOverwriteNegative {
    static int dfb_source() { // DFB-SOURCE: local-overwrite-input
        return 1;
    }

    static void dfb_sink(int value) { } // DFB-SINK: local-overwrite-sink

    static void run() {
        int value = dfb_source();
        value = 0; // DFB-KILL: local-overwrite-clean
        dfb_sink(value);
    }
}

File: cases/taint/scala/local-overwrite-negative/LocalOverwriteNegative.scala
package dataflowbench

object LocalOverwriteNegative {
  def dfb_source(): Int = { // DFB-SOURCE: local-overwrite-input
    1
  }

  def dfb_sink(value: Int): Unit = {} // DFB-SINK: local-overwrite-sink

  def run(): Unit = {
    var value = dfb_source()
    value = 0 // DFB-KILL: local-overwrite-clean
    dfb_sink(value)
  }
}
